src/nvim/msgpack_rpc/server.c

```diff
@@ -224,7 +224,7 @@ int server_start(const char *endpoint)
 /// @param endpoint Address of the server.
 void server_stop(char *endpoint)
 {
-  Server *server;
+  Server *server = NULL;
```

Comment:
Hrmph.  I had assumed that `ga_len` was an unsigned type and didn't even think to look. :-(

I think the fix is far simpler here:

```diff
diff --git a/src/nvim/msgpack_rpc/server.c b/src/nvim/msgpack_rpc/server.c
index 8fb2902..5e3fd10 100644
--- a/src/nvim/msgpack_rpc/server.c
+++ b/src/nvim/msgpack_rpc/server.c
@@ -238,7 +238,7 @@ void server_stop(char *endpoint)
     }
   }

-  if (i == servers.ga_len) {
+  if (i >= servers.ga_len) {
     ELOG("Not listening on %s", addr);
     return;
   }
```

Thanks for looking at it closer though!